pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/tenant/TenantRebalanceResult.java

```diff
@@ -65,4 +215,449 @@ public void setJobId(String jobId) {
+    }
+  }
+
+  /**
+   * Aggregates consuming segment summaries for realtime tables
+   */
+  private static RebalanceSummaryResult.ConsumingSegmentToBeMovedSummary aggregateConsumingSegmentSummary(
+      List<RebalanceSummaryResult.ConsumingSegmentToBeMovedSummary> summaries) {
+    if (summaries.isEmpty()) {
+      return null;
+    }
+
+    int totalNumConsumingSegmentsToBeMoved = 0;
+    // Find the maximum size of offset and age maps across all tables
+    int maxOffsetMapSize = 0;
+    int maxAgeMapSize = 0;
+    for (RebalanceSummaryResult.ConsumingSegmentToBeMovedSummary summary : summaries) {
+      if (summary.getConsumingSegmentsToBeMovedWithMostOffsetsToCatchUp() != null) {
+        maxOffsetMapSize = Math.max(maxOffsetMapSize,
+            summary.getConsumingSegmentsToBeMovedWithMostOffsetsToCatchUp().size());
+      }
+      if (summary.getConsumingSegmentsToBeMovedWithOldestAgeInMinutes() != null) {
+        maxAgeMapSize = Math.max(maxAgeMapSize,
+            summary.getConsumingSegmentsToBeMovedWithOldestAgeInMinutes().size());
+      }
+    }
+
+    // Create maps to store all segments by offset and age
+    Map<String, Integer> allConsumingSegmentsWithOffsets = new HashMap<>();
+    Map<String, Integer> allConsumingSegmentsWithAges = new HashMap<>();
+
+    // Aggregate ConsumingSegmentSummaryPerServer by server name across all tables
+    Map<String, AggregatedConsumingSegmentSummaryPerServer> serverAggregates = new HashMap<>();
+
+    for (RebalanceSummaryResult.ConsumingSegmentToBeMovedSummary summary : summaries) {
+      totalNumConsumingSegmentsToBeMoved += summary.getNumConsumingSegmentsToBeMoved();
+
+      // Add all segments with offsets
+      if (summary.getConsumingSegmentsToBeMovedWithMostOffsetsToCatchUp() != null) {
+        allConsumingSegmentsWithOffsets.putAll(summary.getConsumingSegmentsToBeMovedWithMostOffsetsToCatchUp());
+      }
+
+      // Add all segments with ages
+      if (summary.getConsumingSegmentsToBeMovedWithOldestAgeInMinutes() != null) {
+        allConsumingSegmentsWithAges.putAll(summary.getConsumingSegmentsToBeMovedWithOldestAgeInMinutes());
+      }
+
+      // Aggregate server consuming segment summaries by server name
+      if (summary.getServerConsumingSegmentSummary() != null) {
+        for (Map.Entry<String,
+            RebalanceSummaryResult.ConsumingSegmentToBeMovedSummary.ConsumingSegmentSummaryPerServer> entry
+            : summary.getServerConsumingSegmentSummary()
+            .entrySet()) {
+          String serverName = entry.getKey();
+          RebalanceSummaryResult.ConsumingSegmentToBeMovedSummary.ConsumingSegmentSummaryPerServer serverSummary =
+              entry.getValue();
+
+          serverAggregates.computeIfAbsent(serverName, k -> new AggregatedConsumingSegmentSummaryPerServer())
+              .merge(serverSummary);
+        }
+      }
+    }
+
+    // Keep only top k segments with the highest offsets
```

Comment:
yeah like I mentioned earlier, should we keep a top K per server rather than overall? here K can be smaller like top 3. Same for age